### PR TITLE
Compact Menu

### LINKED ISSUE #1037: Compact Menu
- [x] Menu item height

- [x] Positioning **Should be done in #1452**

- [x] Bug with table row hover when menu is open **Did not find any bugs related to this, probably fixed before**

### PR DESCRIPTION
**Pull request by pomfrida**
Resolves #1037

### CODE REVIEW

File: libraries/core-react/src/components/Menu/MenuList.tsx [hunk 1]
```diff
@@ -26,7 +28,7 @@ const List = styled.ul`
   list-style: none;
   padding: 0;
   margin: 0;
-
+  ${spacingsTemplate(tokens.spacings)}
```
Comment (pomfrida) [inline libraries/core-react/src/components/Menu/MenuList.tsx:31]: `Menu` was missing top and bottom spacing. `Padding: 0` is needed to strip `ul` padding
Screenshot 2021-06-22 at 12 49 16

File: libraries/core-react/src/components/Menu/Menu.tsx [hunk 2]
```diff
@@ -28,15 +28,6 @@ const MenuPaper = styled(Paper)<MenuPaperProps>`
   min-width: fit-content;
   ${bordersTemplate(border)};
   ${({ open }) => css({ visibility: open ? null : 'hidden' })}
-
-  li:first-child {
-    border-top-left-radius: ${border.type === 'border' && border.radius};
-    border-top-right-radius: ${border.type === 'border' && border.radius};
-  }
-  li:last-child {
```
Comment (pomfrida) [inline libraries/core-react/src/components/Menu/Menu.tsx:36]: This border-radius hack is no longer needed after spacing was re-introduced to Menu

File: libraries/core-react/stories/playground/DataTable.stories.tsx [hunk 3]
```diff
@@ -265,8 +265,9 @@ export const CompactDataGrid: Story<TableProps> = () => {
                       color={colors.text.static_icons__tertiary.hex}
                       group="navigation"
                       variant="label"
+                      style={{ height: 16 }}
                     >
-                      <Icon name="folder" />
+                      <Icon name="folder" size={16} />
```
Comment (pomfrida) [inline libraries/core-react/stories/playground/DataTable.stories.tsx:270]: Inline style in `Typography` should be fixed in #1452

File: libraries/core-react/stories/playground/DataTable.stories.tsx [hunk 4]
```diff
@@ -275,6 +276,7 @@ export const CompactDataGrid: Story<TableProps> = () => {
                       color={colors.text.static_icons__tertiary.hex}
                       group="navigation"
                       variant="label"
+                      style={{ height: 12 }}
```
Comment (pomfrida) [inline libraries/core-react/stories/playground/DataTable.stories.tsx:279]: Inline style in Typography should be fixed in #1452

**Review by mimarz — APPROVED** [review]
LGTM 👍

**Pull request merged**